CombineDemo/CombineDemoTests/Publishers/Practice/URLSessionTests.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  URLSessionTests.swift
+//  CombineDemoTests
+//
+//  Created by Manish Rathi on 03/04/2023.
+//
+
+import Foundation
+import Combine
+import XCTest
+
+final class URLSessionTests: XCTestCase {
+    var urlSession: URLSession!
+    var publisher: URLSession.DataTaskPublisher!
+    var cancellables: Set<AnyCancellable>!
+    var isFinishedCalled: Bool!
+    var receivedData: Data?
+    var receivedError: Error?
+
+    override func setUp() {
+        super.setUp()
+
+        let apiURL = URL(string: "https://api.github.com/users/crazymanish")
+        let urlRequest = URLRequest(url: apiURL!)
+        urlSession = URLSession.shared
+        publisher = URLSession.DataTaskPublisher(request: urlRequest, session: urlSession)
```

Comment:
Instead of using real request, please take a look `URLProtocol`, e.g.
https://www.theinkedengineer.com/articles/mocking-requests-using-url-protocol
or `https://www.google.com/search?q=mock+urlprotocol`